Sources/swift-memcache-gsoc-example/Program.swift

```diff
@@ -12,31 +12,46 @@
 //
 //===----------------------------------------------------------------------===//
 
+import Logging
 import NIOCore
 import NIOPosix
+import ServiceLifecycle
 import SwiftMemcache
 
 @main
 struct Program {
-    // Create an event loop group with a single thread
-    static let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    // Use the shared singleton instance of MultiThreadedEventLoopGroup
+    static let eventLoopGroup = MultiThreadedEventLoopGroup.singleton
+    // Initialize the logger
+    static let logger = Logger(label: "memcache")
 
     static func main() async throws {
-        // Instantiate a new MemcachedConnection actor with host, port, and event loop group
-        let memcachedConnection = MemcachedConnection(host: "127.0.0.1", port: 11211, eventLoopGroup: eventLoopGroup)
+        // Instantiate a new MemcacheConnection actor with host, port, and event loop group
+        let memcacheConnection = MemcachedConnection(host: "127.0.0.1", port: 11211, eventLoopGroup: eventLoopGroup)
+
+        // Initialize the ServiceGroupConfiguration
+        let serviceGroupConfiguration = ServiceGroupConfiguration(
+            services: [ServiceGroupConfiguration.ServiceConfiguration(service: memcacheConnection)],
+            gracefulShutdownSignals: [],
+            cancellationSignals: [],
+            logger: self.logger
+        )
+
+        // Initialize the service group
+        let serviceGroup = ServiceGroup(configuration: serviceGroupConfiguration)
```

Comment:
We can use the shorter init here
```suggestion
        // Initialize the service group
        let serviceGroup = ServiceGroup(services: memcacheConnection)
```